addons/mail/static/tests/message/message.test.js

```diff
@@ -2032,3 +2032,33 @@ test("deleted message should not have translate feature", async () => {
         await contains(".dropdown-item:contains('Translate')", { count: 0 });
     }
 });
+
+test("deleting message also removes its link preview", async () => {
+    const pyEnv = await startServer();
+
+    const linkPreviewId = pyEnv["mail.link.preview"].create({
+        og_title: "Delete Test Link",
+        og_description: "Should be removed with the message.",
+        og_type: "article",
+        source_url: "https://www.odoo.com",
+    });
+
+    const channelId = pyEnv["discuss.channel"].create({ name: "DeletePreviewTest" });
+    pyEnv["mail.message"].create({
+        body: "<a href='https://www.odoo.com'>https://www.odoo.com</a>",
+        link_preview_ids: [linkPreviewId],
+        message_type: "comment",
+        model: "discuss.channel",
+        res_id: channelId,
+    });
+
+    await start();
+    await openDiscuss(channelId);
+    await contains(".o-mail-LinkPreviewCard");
+    await contains(".o-mail-Message:contains('https://www.odoo.com')");
+    await click(".o-mail-Message [title='Expand']");
+    await contains(".dropdown-menu");
+    await click(".dropdown-item:contains('Delete')");
```

Comment:
no need for extra checks here
```suggestion
    await click(".o-mail-Message [title='Expand']");
    await click(".o-mail-Message-moreMenu [title='Delete']");
```